src/zombiereborn.cpp

```diff
@@ -817,15 +821,19 @@ void ZR_Hook_ClientPutInServer(CPlayerSlot slot, char const *pszName, int type,
 	if (!pController)
 		return;
 
+	pController->ChangeTeam(g_ZRRoundState == EZRRoundState::POST_INFECTION ? CS_TEAM_T : CS_TEAM_CT);
+
 	// Make sure the round ends if joining an empty server
-	if (ZR_CheckWinConditions(CS_TEAM_CT));
+	if (!ZR_IsTeamAlive(CS_TEAM_CT) && !ZR_IsTeamAlive(CS_TEAM_T) && g_ZRRoundState != EZRRoundState::ROUND_END)
 	{
-		pController->ChangeTeam(CS_TEAM_T);
+		
+		// CONVAR_TODO
+		// use mp_round_restart_delay here
+		g_pGameRules->TerminateRound(5.0f, CSRoundEndReason::GameStart);
```

Comment:
I don't think we will need to obey `mp_round_restart_delay` here, as it's not really a normal round restart. The sooner it happens the better even, 1s or even 0s (if it works well) would be preferable.